### PR TITLE
Feature/login2.0

### PR DESCRIPTION
**Pull request by Ph0tonic**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt [hunk 1]
```diff
@@ -40,7 +42,8 @@ class MapActivityTest {
 
     @Rule
     @JvmField
-    val grantPermissionRule: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION, android.Manifest.permission.ACCESS_FINE_LOCATION)
+    val grantPermissionRule: GrantPermissionRule = GrantPermissionRule
```
Comment (Sporarum) [inline app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:45]: Why not remove it here as well ?

File: app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    @Test
+    fun clickSignInShouldLaunchIntent() {
+        runOnUiThread {
+            Auth.logout()
+            Auth.loggedIn.postValue(false)
+        }
+
+        openDrawer()
+        mUiDevice.wait(Until.hasObject(By.text(getContext().getString(R.string.auth_signin))), TIMEOUT)
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt:84]: Is it really necessary to wait for it to be visible ?

File: app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        }
+
+        openDrawer()
+        runOnUiThread {
+            Auth.email.postValue(FAKE_EMAIL)
+            Auth.name.postValue(FAKE_NAME)
+            Auth.profileImageURL.postValue(FAKE_PROFILE_IMAGE_URL)
+            Auth.loggedIn.postValue(true)
+        }
+        mUiDevice.wait(Until.hasObject(By.text(getContext().getString(R.string.auth_signout))), TIMEOUT)
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt:106]: Is it really necessary to wait for it to be visible ?

File: app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        onView(withId(R.id.nav_user_image)).check(matches(isDisplayed()))
+
+        onView(withText(getContext().getString(R.string.auth_signout))).perform(click())
+        runOnUiThread {
+            Auth.logout()
+            Auth.loggedIn.postValue(false)
+        }
+
+        openDrawer()
+        mUiDevice.wait(Until.hasObject(By.text(getContext().getString(R.string.auth_signin))), TIMEOUT)
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt:119]: Is it really necessary to wait for it to be visible ?

File: app/src/main/java/ch/epfl/sdp/Auth.kt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     *  Allow to launch the connect from a Fragment or anActivity
+     *  Need to override onActivityResult and call Auth.onActivityResult
+     */
+    fun login(fragment: Fragment) {
+        fragment.startActivityForResult(createGoogleSignClient().signInIntent, RC_SIGN_IN)
+    }
+
+    fun login(activity: Activity) {
+        activity.startActivityForResult(createGoogleSignClient().signInIntent, RC_SIGN_IN)
```
Comment (Sporarum) [inline app/src/main/java/ch/epfl/sdp/Auth.kt:48]: Couldn't this and above be replaced by a method that takes a context as argument ?

File: app/src/main/java/ch/epfl/sdp/CentralLocationManager.kt [hunk 6]
```diff
@@ -66,15 +67,15 @@ object CentralLocationManager {
     }
 
     private fun checkPermission(): Boolean {
-        return checkPermission(ACCESS_FINE_LOCATION) && checkPermission(ACCESS_COARSE_LOCATION)
+        return requiredPermissions.all { checkPermissions(it) }
     }
 
-    private fun checkPermission(permission: String): Boolean {
+    private fun checkPermissions(permission: String): Boolean {
```
Comment (Sporarum) [inline app/src/main/java/ch/epfl/sdp/CentralLocationManager.kt:73]: Why change to plural ?
(it only checks a singular permission)

Comment (Ph0tonic) [reply app/src/main/java/ch/epfl/sdp/CentralLocationManager.kt:73]: You're right, I changed the wrong function, I'll fix this.

File: app/src/main/java/ch/epfl/sdp/Auth.kt (hunk 5 above)
Comment (Ph0tonic) [reply app/src/main/java/ch/epfl/sdp/Auth.kt:48]: No, we need an activity or a fragment, it's part of the lifecycle of an activity.

File: app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt (hunk 1 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/MapActivityTest.kt:45]: Yes, I will change this.

File: app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt (hunk 2 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt]: You're right, fixed

File: app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt (hunk 3 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt]: Same

File: app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt (hunk 4 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/LoginNavFragmentTest.kt]: Same

**Pull request merged**